algebird-core/src/main/scala/com/twitter/algebird/BloomFilter.scala

```diff
@@ -150,6 +154,8 @@ case class BFItem(item: String, hashes: BFHash, width: Int) extends BF {
 
   def +(other: String) = this ++ BFItem(other, hashes, width)
 
+  def checkAndAdd(other: String): (BF, ApproximateBoolean) = (this + other, ApproximateBoolean.exact(other == item))
```

Comment:
Sure, that would save an extra hashes calculation when we have a dataset with repeating element.
Rewriting to:
```
  def checkAndAdd(other: String): (BF, ApproximateBoolean) = {
    if (other == item) {
      (this, ApproximateBoolean.exactTrue)
    } else {
      (this + other, ApproximateBoolean.exactFalse)
    }
  }
```